cmd/build.go

```diff
@@ -110,47 +112,73 @@ func Build(ctx context.Context) *cobra.Command {
 	return cmd
 }
 
-func buildV2(m model.ManifestBuild, options build.BuildOptions, args []string) error {
+func keepOnlySelectedServices(service string, manifest *model.ManifestBuild) {
+	for key := range *manifest {
+		if key != service {
+			delete(*manifest, key)
+		}
+	}
+}
+
+func buildV2(manifest model.ManifestBuild, options build.BuildOptions, args []string) error {
 	service := ""
 	if len(args) == 1 {
 		service = args[0]
 	}
 
+	// settings for single build
 	if service != "" {
-		buildInfo, ok := m[service]
+		_, ok := manifest[service]
 		if !ok {
 			return fmt.Errorf("invalid service name: %s", service)
 		}
-		if !okteto.Context().IsOkteto && buildInfo.Image == "" {
-			return fmt.Errorf("image tag is required when context is not okteto")
-		}
+
+		keepOnlySelectedServices(service, &manifest)
 
 		if options.Target != "" {
-			buildInfo.Target = options.Target
+			manifest[service].Target = options.Target
 		}
 		if len(options.CacheFrom) != 0 {
-			buildInfo.CacheFrom = options.CacheFrom
+			manifest[service].CacheFrom = options.CacheFrom
 		}
 		if options.Tag != "" {
-			buildInfo.Image = options.Tag
+			manifest[service].Image = options.Tag
 		}
 
-		opts := build.OptsFromManifest(service, buildInfo, options)
-		opts.Secrets = options.Secrets
-
-		return buildV1(opts, []string{opts.Path})
-	}
-
-	if options.Tag != "" || options.Target != "" || options.CacheFrom != nil || options.Secrets != nil {
+	} else if options.Tag != "" || options.Target != "" || options.CacheFrom != nil || options.Secrets != nil {
 		return fmt.Errorf("flags are not allowed when building services from manifest")
 	}
 
-	for service, buildInfo := range m {
-		if !okteto.Context().IsOkteto && buildInfo.Image == "" {
+	for srv, manifestOptions := range manifest {
+		if !okteto.Context().IsOkteto && manifestOptions.Image == "" {
 			oktetoLog.Errorf("image is required")
```

Comment:
Although this is old code, clarifying which service is missing an image property would be useful for multi service manifests.

```suggestion
			oktetoLog.Errorf("image is required for service %s", srv)
```